### PR TITLE
Rename Vault to Archive

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [X] Code cleanup
- [ ] Build system change
- [X] Documentation change
- [ ] Language translation

### Pull request long description:

The usage of  `Vault` is a poorly chosen name in the context of this project as it has no relation to the services consumed and is no service we provide.
Rename `Vault` to something that is more in line with what this external component really is.

### Changes made:

1. Rename `Vault` to `Archive`
2. Prefix all settings related to s3 storage with `S3_`

### CODE REVIEW

File: deploy/images/db/main.sql [hunk 1]
```diff
@@ -78,15 +78,15 @@ CREATE TABLE local_ega.main (
        submission_file_size                     BIGINT NULL,
        submission_user                          TEXT NOT NULL, -- Elixir ID, or internal user
  
-       -- Vault information
-       vault_file_reference      TEXT,    -- file path if POSIX, object id if S3
-       vault_file_type           storage, -- S3 or POSIX file system
-       vault_file_size           BIGINT,
-       vault_file_checksum       VARCHAR(128) NULL, -- NOT NULL,
-       vault_file_checksum_type  checksum_algorithm,
+       -- archive information
```
Comment (blankdots) [inline deploy/images/db/main.sql]: probably good to keep the capitalisation for consistency :)
```suggestion
       -- Archive information
```

File: lega/ingest.py [hunk 2]
```diff
@@ -85,9 +85,9 @@ def work(fs, channel, data):
         LOG.info(f'[{fs.__class__.__name__}] Moving the rest of {filepath} to {target}')
         target_size = fs.copy(infile, target)  # It will copy the rest only
 
-        LOG.info(f'Vault copying completed. Updating database')
+        LOG.info(f'archive copying completed. Updating database')
```
Comment (blankdots) [inline lega/ingest.py]: capitalisation :+1: 
```suggestion
        LOG.info(f'Archive copying completed. Updating database')
```

File: deploy/images/db/main.sql [hunk 3]
```diff
@@ -198,7 +198,7 @@ CREATE FUNCTION finalize_file(inpath        local_ega.files.inbox_path%TYPE,
 	-- 		inbox_path = inpath AND
 	-- 		status <> 'COMPLETED')
 	-- THEN
-	--    RAISE EXCEPTION 'Vault file not in completed state for stable_id: % ', sid;
+	--    RAISE EXCEPTION 'archive file not in completed state for stable_id: % ', sid;
```
Comment (blankdots) [inline deploy/images/db/main.sql]: capitalisation :+1: 
```suggestion
	--    RAISE EXCEPTION 'Archive file not in completed state for stable_id: % ', sid;
```

File: docs/connection.rst [hunk 4]
```diff
@@ -57,9 +57,9 @@ use the stub implementation of CentralEGA and the following queues, per
 +=================+=====================================+
 | files           | Trigger for file ingestion          |
 +-----------------+-------------------------------------+
-| archived        | The file is in the vault            |
+| archived        | The file is in the archive            |
```
Comment (blankdots) [inline docs/connection.rst]: this is a malformed table, see error from Sphinx:
```
/home/stenegru/csc/dev/EGA-LocalEGA/docs/connection.rst:55: WARNING: Malformed table.

+-----------------+-------------------------------------+
| Name            | Purpose                             |
+=================+=====================================+
| files           | Trigger for file ingestion          |
+-----------------+-------------------------------------+
| archived        | The file is in the archive            |
+-----------------+-------------------------------------+
| qc              | The file is "verified" in the archive |
|                 | and Quality Controllers can execute |
+-----------------+-------------------------------------+
```

to replicate this (must have `sphinx` and `sphinx_rtd_theme` installed):
```
cd docs
make html
```

should look like:
```
+-----------------+---------------------------------------+
| Name            | Purpose                               |
+=================+=======================================+
| files           | Trigger for file ingestion            |
+-----------------+---------------------------------------+
| archived        | The file is in the archive            |
+-----------------+---------------------------------------+
| qc              | The file is "verified" in the archive |
|                 | and Quality Controllers can execute   |
+-----------------+---------------------------------------+
```

File: lega/utils/storage.py [hunk 5]
```diff
@@ -15,17 +15,17 @@
 
 
 class FileStorage():
-    """Vault storage on disk and related I/O."""
+    """archive storage on disk and related I/O."""
```
Comment (blankdots) [inline lega/utils/storage.py]:
```suggestion
    """Archive storage on disk and related I/O."""
```

File: lega/utils/storage.py [hunk 6]
```diff
@@ -204,17 +204,17 @@ def _fetch(self, start, end, max_attempts=10):
 
 
 class S3Storage():
-    """Vault S3 object storage and related I/O."""
+    """archive S3 object storage and related I/O."""
```
Comment (blankdots) [inline lega/utils/storage.py]:
```suggestion
    """Archive S3 object storage and related I/O."""
```

**Comment by dtitov** [comment]
@silverdaz 

It was **completely** the same situation with PR https://github.com/EGA-archive/LocalEGA/pull/5

You made a bunch of renamings there which didn't affect functionality at all, neither they fixed anything. `s3` became `vault`, `docker` folder became `deploy` folder, `id-mapper` became `finalizer` (which now, by the ways, causes some issues, as you've seen in Slack). 

So to me, these two PRs are of the same nature: purely non-functional. But since we accepted that PR, we should probably accept this one as well.

**Comment by silverdaz** [comment]
I'm not against the renaming.
It's just a name. If it helps some people resolve a confusion, then so be it.
Maybe with the risk that this new name will introduce other confusions, somehow.

I don't mind.

But I don't see the issue on the Project Board, so I got confused to why this showed up now and here, and that we have to deal with this.

That said: I think you need @jbygdell to resolve some conflicts after the rebasing.

**Comment by jbygdell** [comment]
Rebased to match master, travis fails due to minio issues that will be fixed in #33

**Pull request merged**